### ISSUE TITLE
Some misc contribute settings suggestions

### ISSUE THREAD

**Issue by N1ck01** [allo-/firefox-profilemaker#267]
Some misc contribute settings suggestions for the project:

Setting: browser.compactmode.show: true
Title: Show "Compact" Density
Description: Show hidden "Compact" Density option in Customize Toolbar page that reduces spacing on toolbars.
https://www.askvg.com/tip-restore-compact-mode-density-option-in-firefox-customize-window/

Setting: extensions.unifiedExtensions.enabled: false
Title: Hide "Extensions" Button
Description: Hide dedicated "Extensions" button from toolbar.
https://www.askvg.com/disable-or-remove-extensions-button-from-firefox-toolbar/

Setting: browser.newtabpage.activity-stream.topSitesRows: #
Title: Add Top Sites Rows For Shortcuts
Description: Add more Top Sites rows beyond the default max of 4 rows for more Shortcuts on the New Tab page.
https://www.ghacks.net/2018/05/17/how-to-add-more-top-sites-rows-to-firefox/

Setting: accessibility.force_disabled: 1
Title: Disable Accessibility Service
Description: Accessibility Service should only be enabled if you use Firefox through any type of physical impairment assistive software, otherwise it should be disabled to stop third party applications or malware from monitoring your web surfing activity, to improve browsing performance, and to improve stability.
https://support.mozilla.org/en-US/kb/accessibility-services

Setting: browser.download.start_downloads_in_tmp_dir: true
Title: Start Downloads In TEMP Folder
Description: Initially put downloads in the OS temp folder instead of the user configured download folder. Files opened from the “what should Firefox do with this file” dialog, or set to open in helper applications automatically, will stay in this temp folder. Files saved (not opened as previously mentioned) will be put into the user configured download folder.
https://www.askvg.com/tip-force-firefox-to-start-downloads-in-temp-folder-again/

Setting: middlemouse.paste: true
Title: Enable Mouse Middle Click To Paste
Description: Enable mouse middle click to paste copied clipboard text into the address bar, search bar, or any text field. (Enabling still allows mouse middle click to open links in a new tab.)
https://www.askvg.com/how-to-enable-secret-middle-click-to-paste-clipboard-content-feature-in-mozilla-firefox/

**Comment by allo-** [comment]
Hello and thank you for the suggestions. I think most of them are not for the main profile, but are useful for additional profiles (#71) as they are not security/privacy relevant.

- Title: Show "Compact" Density: UI Preference. Also available via GUI when you select "Customize Toolbar"
- Title: Hide "Extensions" Button - UI. I am also not sure about the side effects. I hid the button with userCSS before and then there were extensions that were unreachable without the button.
- Setting: browser.newtabpage.activity-stream.topSitesRows - UI
- **Title: Disable Accessibility Service** - That one could be relevant, but I wonder if one needs `force_disabled` or if Firefox first asks for permission when a program wants to act as accessibility service. When there is a confirmation dialog, it is also no setting for the main profile (but e.g. for a more paranoid profile #50).
- Setting: browser.download.start_downloads_in_tmp_dir: true - User Preference and I would think not worth to ask every user using the main profile what's their preference on that.
- Setting: middlemouse.paste: true - Platform specific defaults. Under Linux middlemouse paste is default and you rather need to take care to disable middlemouse scroll.

I leave this open as all settings are interesting for other profiles, but I don't think I want them in the main profile, as there is a limit how many questions a user is willing to answer before blindly accepting defaults, so I try to have the relevant settings in the main profile. See the linked issues above for the plans how and where to add additional settings.

**Comment by allo-** [comment]
The unifiedExtensions setting doesn't seem to have an effect here. Maybe it was a setting in the first release that is now already removed?

**Comment by N1ck01** [comment]
Unfortunately with Firefox 111 the unifiedExtensions setting has been removed so the "Extensions" button is now being permanently forced on the UI.
https://www.reddit.com/r/firefox/comments/11rrl0b/unable_to_remove_unified_extensions_button_after/

Just to add to a previous reply, the "Compact" Density UI preference doesn't actually show in "Customize Toolbar" until its secret setting is enabled to show it as a selection, the topSitesRows setting within the UI only goes up to 4 rows so that setting is needed to add more rows if wanted, the tmp_dir setting used to be how Firefox handled downloads (In older versions of Firefox, the browser used to keep the downloaded files in system’s TEMP folder until the user decided to save the file on hard disk. If the user didn’t want to save the file, it was automatically deleted from the TEMP folder.) which could be a privacy concern, and the middlemouse.paste setting works on Windows if enabled (a different setting to copy on highlight doesn't).
But as previously stated, these settings are not security relevant so it is just user preference.

**Comment by allo-** [comment]
> Just to add to a previous reply, the "Compact" Density UI preference doesn't actually show in "Customize Toolbar" until its secret setting is enabled to show it as a selection
I click on the toolbar (or menu bar if I don't find empty space to click in the toolbar) and when I select customize I have the density dropdown next to the settings like showing the titlebar. Isn't that the same?

I find all these settings reasonable, and I really need the time to implement #71 and then merge all suggestions for such improvements. Then every user could choose themself how many options they want to decide and one could possibly even split off a few from the current profile to a second profile that can be selected together with the default profile to have all options.

**Comment by N1ck01** [comment]
Yes that is the way to get to the Density dropdown, but I think I understand the confusion now. The dropdown is labeled "Density" no matter what option is selected within the dropdown menu, but there's an option that doesn't actually show in that Density dropdown menu by default (the "_Compact (not supported)_" option) unless you enable the config setting for it to show within the dropdown menu, and the option just makes the toolbar UI more compact than the other Density options. Hope that helps to clarify.
![Density 1](https://user-images.githubusercontent.com/94124610/226184520-6ea1b26b-77c1-4525-9780-86d3ee266b96.png)

I'll say it was my mistake for suggesting these misc settings because what makes me really like this tool to begin with is that it's clean, simple, and specifically tailored for security and privacy without unnecessary "bloat". These settings are more suited for a separate profile configurator. Sorry for any hassle.

**Comment by allo-** [comment]
Oh, I see the point. And as said, it is planned, but may take some time. I imagine replacing the dropdown for the profiles with a multi-select box and then you can combine what settings you're want by selecting the profiles that you are interested in.
